frontend/src/components/App.tsx

```diff
@@ -0,0 +1,116 @@
+import * as React from 'react';
+import { useContext, useEffect, useState } from 'react';
+import axios, { AxiosResponse } from 'axios';
+import { HapiPractitionersResponse } from '../types/Hapi.types';
+import PractitionerFeed from './PractitionerFeed';
+import { HapiPractitionersFetchRequestUrl } from '../constants';
+import { AppContext } from '../AppContext';
+import { NormalizedPractitioner } from '../utils/helpers';
+import { Input, Button, CircularProgress } from '@mui/material';
+
+function App() {
+  const appContext = useContext(AppContext);
+  const [practitioners, setPractitioners] = useState<NormalizedPractitioner[]>([]);
+  const [searchTerm, setSearchTerm] = useState('');
+
+  useEffect( () => {
+    let searchUrl = HapiPractitionersFetchRequestUrl;
+
+    if (!appContext?.state.fetched || searchTerm.length === 0) {
```

Comment:
Hurray for not making the call again unnecessarily! 

Curious what you think of SWR or React Query as alternatives for things like this? I get not including them yet, so not a criticism.